src/language_features/document_symbol.rs

```diff
@@ -216,14 +235,41 @@
     ctx: &Context,
 ) -> String {
     fn format_symbol_at_depth<'a, T: Symbol<T>>(
-        output: &mut Vec<(String, String, &'a str)>,
+        output: &mut Vec<(String, String, String)>,
         items: &'a [T],
         meta: &EditorMeta,
         server: &ServerSettings,
         ctx: &Context,
         depth: usize,
+        prefix: &Vec<bool>,
     ) {
-        for symbol in items {
+        let length = items.len();
+        let is_root = depth == 0;
+        for (index, symbol) in items.iter().enumerate() {
+            let is_last = index + 1 == length;
+
+            let mut new_prefix = prefix.clone();
+            new_prefix.push(is_last);
+
+            let hierarchy = if is_root {
+                "".to_string()
+            } else {
+                let last_hierarchy_symbol = if is_last {
+                    Tree::LastChild
+                } else {
+                    Tree::Child
+                };
+                let prefixing_hierarchy_symbols = prefix
+                    .iter()
+                    .rev()
+                    .take(depth - 1)
+                    .map(|flag| if *flag { Tree::Empty } else { Tree::Pipe })
+                    .rev()
```

Comment:
`.rev().take().rev()` can be replaced by `skip(1)`.
It makes sense because we don't want to indent top-level symbols. Could also try to not push if root but that'd probably be uglier.

Reply:
Done